components/layout/mediaQuery/src/index.js

```diff
@@ -57,10 +68,19 @@ const LayoutMediaQueryFactory = BREAKPOINTS =>
       if (!shallowEqual(result, params)) this.setState({params: result})
     }
 
+    shouldComponentUpdate(_, nextState) {
+      return !shallowEqual(this.state.params, nextState.params)
```

Comment:
This hook is not required as long while you are using a PureComponent.
https://reactjs.org/docs/shallow-compare.html

In any case.. im not pretty sure about that we need a pure component on this piece. Why did you changed the stateless version for a pure component?

Reply:
I had remove the pure component. And left the shouldComponentUpdate. The PureComponent only didnt works.